bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+                        tmpString = "";
+                        for (int i = 1; i <= zoneNum; i++) {
+                            switch (i) {
+                                case 1:
+                                    tmpString = "\"main\"";
+                                    break;
+                                case 2:
+                                    tmpString = tmpString + ", \"zone2\"";
+                                    break;
+                                case 3:
+                                    tmpString = tmpString + ", \"zone3\"";
+                                    break;
+                                case 4:
+                                    tmpString = tmpString + ", \"zone4\"";
+                                    break;
+                            }
+                        }
+                        json = "{\"group_id\":\"" + groupId + "\", \"zone\":[" + tmpString + "]}";
+                        logger.info("setClientInfo json: {}", json);
+                        httpResponse = setClientInfo(this.host, json);
+                        httpResponse = startDistribution(mclinkSetupServer);
+                    }
+                    break;
+                case CHANNEL_UNLINKMCSERVER:
+                    if (command.equals(OnOffType.ON)) {
+                        json = "{\"group_id\":\"\"}";
+                        httpResponse = setServerInfo(this.host, json);
+                        updateState(channelUID, OnOffType.OFF);
+                    }
+                    break;
+                case CHANNEL_RECALLSCENE:
+                    recallScene(command.toString(), zone);
+                    break;
+                case CHANNEL_REPEAT:
+                    setRepeat(command.toString());
+                    break;
+                case CHANNEL_SHUFFLE:
+                    setShuffle(command.toString());
+                    break;
+            } // END Switch Channel
+        }
+    }
+
+    @Override
+    public void initialize() {
+        thingLabel = thing.getLabel();
+        logger.info("YXC - Start initializing! - {}", thingLabel);
+        this.config = getConfigAs(YamahaMusiccastConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        if (config.configHost != null) {
+            this.host = config.configHost;
+        }
+        if (!this.host.equals("")) {
+            zoneNum = getNumberOfZones(this.host);
+            logger.info("YXC - Zones found: {} - {}", zoneNum, thingLabel);
+
+            if (zoneNum > 0) {
+                refreshOnStartup();
+                keepUdpEventsAliveTask = scheduler.scheduleWithFixedDelay(this::keepUdpEventsAlive, 5, 300,
+                        TimeUnit.SECONDS);
+                logger.info("YXC - Start Keep Alive UDP events (5 minutes - {}) ", thingLabel);
+
+                updateStatus(ThingStatus.ONLINE);
+                logger.info("YXC - Finished initializing! - {}", thingLabel);
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Reply:
Will be availalbe in next push

Reply:
Done